rules/aip0122/embedded_resource.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0122
+
+import (
+	"fmt"
+
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+)
+
+var embeddedResource = &lint.MessageRule{
+	Name:   lint.NewRuleName(122, "embedded-resource"),
+	OnlyIf: utils.IsResource,
+	LintMessage: func(m *desc.MessageDescriptor) []lint.Problem {
+		var problems []lint.Problem
+		for _, f := range m.GetFields() {
+			mt := f.GetMessageType()
+			if mt == nil || !utils.IsResource(mt) {
```

Comment:
before I forget: can you add the check for a revision resource? I think using the message type may be sufficient. 

This probably isn't the right place to put the guard condition (maybe updating OnlyIf?)

Reply:
I first considered the `OnlyIf`, but thought we should be more principled and only exempt valid resource snapshots in revision resources, but not allow another random resource to be embedded in the revision resource. Working on that now.

Reply:
Done!